### PR TITLE
Pt 158185357 Sophia data decode endpoint

### PR DESCRIPTION
**Pull request by happi**
See also 
https://github.com/aeternity/protocol/pull/123

### CODE REVIEW

File: apps/aecontract/test/aecontract_SUITE.erl [hunk 1]
```diff
@@ -461,7 +461,7 @@ sophia_identity(_Cfg) ->
     %% Remote calling the identity contract
     IdC   = ?call(create_contract, Acc1, identity, {}),
     RemC  = ?call(create_contract, Acc1, remote_call, {}, #{amount => 100}),
-    42    = ?call(call_contract,   Acc1, IdC, main, word, 42),
+    42    = ?call(call_contract,   Acc1, IdC, main, word, {42}),
```
Comment (UlfNorell) [inline apps/aecontract/test/aecontract_SUITE.erl]: Change `call_contract` to take a list of arguments?

File: apps/aesophia/src/aeso_data.erl [hunk 2]
```diff
@@ -61,19 +65,36 @@ binary_to_words(<<N:256,Bin/binary>>) ->
...
+-ifdef(TEST).
+print_heap(Heap) ->
+    print_heap(0, Heap).
+print_heap(_, <<>>) -> ok;
+print_heap(A, <<W:256, Rest/binary>>) ->
+    io:format("~p : ~p~n", [A,W]),
+    print_heap(A+32, Rest);
+print_heap(A, <<B:8, Rest/binary>>) ->
+    io:format("~p : [~p]~n", [A,B]),
+    print_heap(A+1, Rest).
```
Comment (UlfNorell) [inline apps/aesophia/src/aeso_data.erl]: See also `aeso_test_utils:dump_words/1`.

File: apps/aecontract/test/aecontract_SUITE.erl (hunk 1 above)
Comment (happi) [reply apps/aecontract/test/aecontract_SUITE.erl]: It would require a bit more redesign.

File: config/swagger.yaml [hunk 3]
```diff
@@ -2006,6 +2006,33 @@ paths:
           description: Invalid contract
           schema:
             $ref: '#/definitions/Error'
+  /contract/decode-data:
```
Comment (lucafavatella) [inline config/swagger.yaml]: Worth a line in release notes? Though optional.

File: apps/aehttp/test/aehttp_integration_SUITE.erl [hunk 4]
```diff
@@ -579,32 +657,35 @@ block_by_hash(_Config) ->
...
     {ok, 200, #{<<"pub_key">> := MinerAddress}} = get_miner_pub_key(),
     {ok, MinerPubkey} = aec_base58c:safe_decode(account_pubkey, MinerAddress),
+    SophiaCode = <<"contract Identity = function main (x:int) = x">>,
+    {ok, 200, #{<<"bytecode">> := Code}} = get_contract_bytecode(SophiaCode),
 
     % contract_create_tx positive test
-    Code = <<"0x600035807f00000000000000000000000000000000000000000000000000000"
-	     "00000000000146200002c57005b6020356200003a9062000043565b6000526020"
-	     "6000f35b8090509056">>,
+    %% Code = <<"0x600035807f00000000000000000000000000000000000000000000000000000"
```
Comment (lucafavatella) [inline apps/aehttp/test/aehttp_integration_SUITE.erl]: So this can be deleted?

File: config/swagger.yaml [hunk 5]
```diff
@@ -2987,6 +3014,18 @@ definitions:
     properties:
       calldata:
         type: string
+  SophiaBinaryData:
+    type: object
+    properties:
+      type:
+        type: string
+      data:
+        type: string
```
Comment (lucafavatella) [inline config/swagger.yaml]: Shall both type and data be required?

File: config/swagger.yaml [hunk 6]
```diff
@@ -2987,6 +3014,18 @@ definitions:
...
+  SophiaJsonData:
+    type: object
+    properties:
+      data:
+        type: object
```
Comment (lucafavatella) [inline config/swagger.yaml]: Maybe data shall be required so client knows can rely on it?

File: config/swagger.yaml [hunk 7]
```diff
@@ -2987,6 +3014,18 @@ definitions:
     properties:
       calldata:
         type: string
+  SophiaBinaryData:
+    type: object
+    properties:
+      type:
```
Comment (lucafavatella) [inline config/swagger.yaml]: type or sophia-type ?

Comment (lucafavatella) [inline config/swagger.yaml]: sophia-type ?

File: config/swagger.yaml (hunk 3 above)
Comment (happi) [reply config/swagger.yaml]: Done

File: apps/aehttp/test/aehttp_integration_SUITE.erl (hunk 4 above)
Comment (happi) [reply apps/aehttp/test/aehttp_integration_SUITE.erl]: Done

**Pull request merged**